### ISSUE TITLE
"Open Playlist Error" Dialogue Flagged Upon Opening a Broken Playlist Containing a Special Character in a Track Path

### ISSUE THREAD

**Issue by touwys** [Borewit/listFix#266]
STEPS:

 1. **Prerequisite**: For editing, choose a broken playlist that contains at least one track which is located in a path that contains a special Windows character. 
 
    > **Note**: _In this particular instance, the path contains the special dot • character that is produced using the Windows keyboard combination `Alt+7`. This character stands applied to numerous music library folder names, and consequently happens to be in the path._
 
2. Double-click the playlist to open it in the Playlist Editor.

3. The 'Open Playlist Error' dialogue gets flagged. (Screenshot attached.)

[Screenshot: 'Open Playlist Error'](https://github.com/Borewit/listFix/assets/65730626/f25ee481-7927-4319-a374-9e9178a73aaa)

[Screenshot: Example of • Character in Folder Names](https://github.com/Borewit/listFix/assets/65730626/e9c4a243-ed7f-46e6-a683-4412d9e7f009)

_Device: PC
OS: Windows 7x64 Ultimate
listFIX(): 2.10.0 "Portable"_

**Comment by Borewit** [comment]
I assume the "•" is the [Unicode bullet character: U+2022](https://en.wikipedia.org/wiki/Bullet_(typography)). That required a playlist format which supports Unicode characters. [M3U](https://en.wikipedia.org/wiki/M3U) does not support [Unicode](https://en.wikipedia.org/wiki/Unicode), [M3U8](https://en.wikipedia.org/wiki/M3U#M3U8) does support Unicode (that's the difference).

![image](https://github.com/Borewit/listFix/assets/23255260/769c397a-32bb-408d-8e3d-f7ac63ea46a6)

Handling of such impossible conversions (causing data loss) could probably be better.

**Comment by touwys** [comment]
Thank you, @Borewit :

> I assume the "•" is the [Unicode bullet character: U+2022](https://en.wikipedia.org/wiki/Bullet_(typography)).

According to AI, it is the same as "Alt+7" Windows Key Code.

> Errors trying to make impossible conversions (causing data loss) could probably be better.

Do you mean that it is probably better to convert all the current m3u playlists to the m3u8 format so that the special characters could well be read within listFix?

**Comment by Borewit** [comment]
> Do you mean that it is probably better to convert all the current m3u playlists to the m3u8 format so that the special characters could well be read within listFix?

That is true, but that is not what I meant, corrected my unreadable sentence.
You can consider `m3u8` as an improved version for `m3u`. `m3u` does not support unicode, but which non-unicode character sets it does support is not defined. 

I once had media player [Dune HD](https://www.dune-hd.com/), which did not understand the `m3u8` extension, only `m3u`. The funny thing was that they implemented the M3U with UTF-8. Reading M3U with UTF-8 text encoding is M3U8. So saving the playlist as `m3u8`, and then renaming the extension to `m3u` was the trick that make that media player to read path with special characters.

But what I meant is that listFix() could theoretically be improved to warn the user when he tries to save data which does not fit in the target format. Yet due to the fact that it is not even clear what does and what does not fit in `.m3u`, so a solution on the best known convention is the maximum achievable.

**Comment by touwys** [comment]
Thank you for the explanation. It also lifts some of the fogginess I may have had, prior.

Now looking for a mass m3u to m3u8 playlist converter. I know of two apps, which I have, that are capable of _exporting_ the m3u playlists to m3u8 format but, pending further investigation, I'm aiming for something decidedly easier to use.

**Comment by Borewit** [comment]
I believe this one is resolved
